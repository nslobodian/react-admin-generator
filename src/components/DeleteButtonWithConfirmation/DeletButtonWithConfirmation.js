import React from 'react'
import FaClose from 'react-icons/lib/fa/close'
import { show } from 'redux-modal/lib/actions'
import withHandlers from 'recompose/withHandlers'
import connect from 'react-redux/lib/connect/connect'

export const DeleteButtonWithConfirmation = withHandlers({
  showDeleteModalHandler: ({ onDelete, deletingId, showDeleteModal, item, deleteMessage }) => () => {
    showDeleteModal(() => onDelete(deletingId), item, deleteMessage)
  },
})(({ showDeleteModalHandler, isButton, className }) =>
  <div onClick={showDeleteModalHandler}>
    {isButton
      ? <button className={`btn btn-sm btn-danger ${className}`}>Delete</button>
      : <span><FaClose className='CloseButton' /></span>
    }
  </div>
)

export const mapActionCreators = {
  showDeleteModal: (handleDelete, item, deleteMessage) => show('confirmDeleteModal', { handleDelete, item, deleteMessage }),
}

export default connect(() => ({}), mapActionCreators)(DeleteButtonWithConfirmation)
