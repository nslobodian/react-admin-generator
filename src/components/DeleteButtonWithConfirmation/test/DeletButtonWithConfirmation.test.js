import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { DeleteButtonWithConfirmation, mapActionCreators } from '../DeletButtonWithConfirmation'

describe('(Components) DeleteButtonWithConfirmation', () => {
  it('Should render DeleteButtonWithConfirmation properly', () => {
    const props = {
      onDelete: jest.fn(),
      deletingId: 1,
      showDeleteModal: jest.fn(),
      item: 'item',
      showDeleteModalHandler: jest.fn(),
      isButton: true,
    }
    const wrapper = mount(
      <DeleteButtonWithConfirmation
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render DeleteButtonWithConfirmation properly without some attributes', () => {
    const props = {
      onDelete: jest.fn(),
      deletingId: 1,
      showDeleteModal: jest.fn(),
      item: 'item',
      showDeleteModalHandler: jest.fn(),
      isButton: false,
    }
    const wrapper = mount(
      <DeleteButtonWithConfirmation
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render mapActionCreators', () => {
    const handleDelete = jest.fn();
    const item = 'item';
    const props = {
      showDeleteModal: jest.fn(),
      item: 'item',
      handleDelete: jest.fn(),
    }
    const wrapper = shallow(
      <mapActionCreators
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
