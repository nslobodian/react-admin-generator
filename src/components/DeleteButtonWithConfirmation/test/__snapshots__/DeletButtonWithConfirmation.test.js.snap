// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) DeleteButtonWithConfirmation Should render DeleteButtonWithConfirmation properly 1`] = `
<_class
  deletingId={1}
  isButton={true}
  item="item"
  onDelete={[Function]}
  showDeleteModal={[Function]}
  showDeleteModalHandler={[Function]}
>
  <div
    onClick={[Function]}
  >
    <button
      className="btn btn-sm btn-danger undefined"
    >
      Delete
    </button>
  </div>
</_class>
`;

exports[`(Components) DeleteButtonWithConfirmation Should render DeleteButtonWithConfirmation properly without some attributes 1`] = `
<_class
  deletingId={1}
  isButton={false}
  item="item"
  onDelete={[Function]}
  showDeleteModal={[Function]}
  showDeleteModalHandler={[Function]}
>
  <div
    onClick={[Function]}
  >
    <span>
      <FaClose
        className="CloseButton"
      >
        <IconBase
          className="CloseButton"
          viewBox="0 0 40 40"
        >
          <svg
            className="CloseButton"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
              />
            </g>
          </svg>
        </IconBase>
      </FaClose>
    </span>
  </div>
</_class>
`;

exports[`(Components) DeleteButtonWithConfirmation Should render mapActionCreators 1`] = `
<mapActionCreators
  handleDelete={[Function]}
  item="item"
  showDeleteModal={[Function]}
/>
`;
